scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1521,6 +1521,9 @@ private void sendMessage(final RawData message) throws HandshakeException {
 		DTLSSession session = connection.getEstablishedSession();
 		SessionTicket ticket = connection.getSessionTicket();
 		if (session == null && ticket == null) {
+			if (config.isServerOnly()) {
+				message.onError(new EndpointMismatchException());
```

Comment:
Using EndpointMismatchException sounds strange to me.
(If we don't find a more appropriate exception, we should at least add a String explanation to the exception)